lib/authz/permissions_test.go

```diff
@@ -451,3 +444,90 @@ type fakeCtxChecker struct {
 func (c *fakeCtxChecker) GetAccessState(_ types.AuthPreference) services.AccessState {
 	return c.state
 }
+
+type testClient struct {
+	services.ClusterConfiguration
+	services.Trust
+	services.Access
+	services.Identity
+	types.Events
+}
+
+func newTestResources(t *testing.T) (*testClient, *services.LockWatcher, Authorizer) {
+	backend, err := memory.New(memory.Config{})
+	require.NoError(t, err)
+
+	clusterConfig, err := local.NewClusterConfigurationService(backend)
+	require.NoError(t, err)
+	caSvc := local.NewCAService(backend)
+	accessSvc := local.NewAccessService(backend)
+	identitySvc := local.NewIdentityService(backend)
+	eventsSvc := local.NewEventsService(backend)
+
+	testClient := &testClient{
```

Comment:
nit: the `testClient` variable shadows the `testClient` type. Tweaking one of the names would be better.